keystore/admin.go

```diff
@@ -0,0 +1,201 @@
+package keystore
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/ed25519"
+	"crypto/rand"
+	"fmt"
+	"maps"
+	"time"
+
+	"golang.org/x/crypto/curve25519"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/smartcontractkit/chainlink-common/keystore/internal"
+)
+
+var (
+	ErrKeyAlreadyExists   = fmt.Errorf("key already exists")
+	ErrInvalidKeyName     = fmt.Errorf("invalid key name")
+	ErrKeyNotFound        = fmt.Errorf("key not found")
+	ErrUnsupportedKeyType = fmt.Errorf("unsupported key type")
+)
+
+type CreateKeysRequest struct {
+	Keys []CreateKeyRequest
+}
+
+type CreateKeyRequest struct {
+	KeyName string
+	KeyType KeyType
+}
+
+type CreateKeysResponse struct {
+	Keys []CreateKeyResponse
+}
+
+type CreateKeyResponse struct {
+	KeyInfo KeyInfo
+}
+
+type DeleteKeysRequest struct {
+	KeyNames []string
+}
+
+type DeleteKeysResponse struct{}
+
+type ImportKeysRequest struct {
+	Keys []ImportKeyRequest
+}
+
+type ImportKeyRequest struct {
+	KeyName string
+	KeyType KeyType
+	Data    []byte
+}
+
+type ImportKeysResponse struct{}
+
+type ExportKeyParam struct {
+	KeyName string
+	Enc     EncryptionParams
+}
+
+type ExportKeysRequest struct {
+	Keys []ExportKeyParam
+}
+
+type ExportKeysResponse struct {
+	Keys []ExportKeyResponse
+}
+
+type ExportKeyResponse struct {
+	KeyName string
+	Data    []byte
+}
+
+type SetMetadataRequest struct {
+	Updates []SetMetadataUpdate
+}
+
+type SetMetadataUpdate struct {
+	KeyName  string
+	Metadata []byte
+}
+
+type SetMetadataResponse struct{}
+
+type Admin interface {
+	CreateKeys(ctx context.Context, req CreateKeysRequest) (CreateKeysResponse, error)
+	DeleteKeys(ctx context.Context, req DeleteKeysRequest) (DeleteKeysResponse, error)
+	ImportKeys(ctx context.Context, req ImportKeysRequest) (ImportKeysResponse, error)
+	ExportKeys(ctx context.Context, req ExportKeysRequest) (ExportKeysResponse, error)
+	SetMetadata(ctx context.Context, req SetMetadataRequest) (SetMetadataResponse, error)
+}
+
+func ValidKeyName(name string) error {
+	if name == "" {
+		return fmt.Errorf("key name cannot be empty")
+	}
+	// Just a sanity bound.
+	if len(name) > 1_000 {
+		return fmt.Errorf("key name cannot be longer than 1000 characters")
+	}
+	return nil
+}
+
+func (ks *keystore) CreateKeys(ctx context.Context, req CreateKeysRequest) (CreateKeysResponse, error) {
+	ks.mu.Lock()
+	defer ks.mu.Unlock()
+
+	ksCopy := maps.Clone(ks.keystore)
+	var responses []CreateKeyResponse
+	for _, keyReq := range req.Keys {
+		if _, ok := ksCopy[keyReq.KeyName]; ok {
+			return CreateKeysResponse{}, fmt.Errorf("%w: %s", ErrKeyAlreadyExists, keyReq.KeyName)
+		}
+		if err := ValidKeyName(keyReq.KeyName); err != nil {
+			return CreateKeysResponse{}, fmt.Errorf("%w: %s", ErrInvalidKeyName, err)
+		}
+		switch keyReq.KeyType {
```

Comment:
still some more key types to add (starknet) but will do that in a follow up as this is already getting large